From 2fbe34703fc8320327792d341387d8ab65e819c4 Mon Sep 17 00:00:00 2001
From: Ben Whitten <ben.whitten@lairdtech.com>
Date: Thu, 20 Apr 2017 12:08:05 +0100
Subject: [PATCH 2/3] laird: Allow connection to IPv6 address

AF_UNSPEC allows connection to both IPv4 and 6 but fails at localhost,
as we have no local network server its not a problem.

Bug 10956
---
 lora_pkt_fwd/src/lora_pkt_fwd.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lora_pkt_fwd/src/lora_pkt_fwd.c b/lora_pkt_fwd/src/lora_pkt_fwd.c
index ae94c84..35287d4 100644
--- a/lora_pkt_fwd/src/lora_pkt_fwd.c
+++ b/lora_pkt_fwd/src/lora_pkt_fwd.c
@@ -1123,7 +1123,9 @@ int main(void)
 
     /* prepare hints to open network sockets */
     memset(&hints, 0, sizeof hints);
-    hints.ai_family = AF_INET; /* WA: Forcing IPv4 as AF_UNSPEC makes connection on localhost to fail */
+    /* Using AF_UNSPEC to allow both IPv4 and 6 addressed
+     * localhost will fail but no network server locally */
+    hints.ai_family = AF_UNSPEC;
     hints.ai_socktype = SOCK_DGRAM;
 
     /* look for server address w/ upstream port */
-- 
2.17.1

